src/main/java/jpa/dao/CompanyDao.java

```diff
@@ -0,0 +1,75 @@
+package jpa.dao;
+
+import jpa.entity.Companies;
+import lombok.extern.log4j.Log4j;
+
+import javax.persistence.EntityManager;
+
+@Log4j
+public class CompanyDao {
+    private EntityManager entityManager;
+
+    public CompanyDao(EntityManager entityManager) {
+        this.entityManager = entityManager;
+    }
+
+    public boolean insertCompany(Companies company) {
+        try {
+            entityManager.getTransaction().begin();
+            entityManager.persist(company);
+            entityManager.getTransaction().commit();
+            return true;
+        } catch (RuntimeException e) {
+            if (entityManager != null) {
```

Comment:
Looks like a duplicate. Look at other methods and move this code into a separate method and reuse.